### PR TITLE
Complete request and response parsing for Route service

### PR DESCRIPTION
**Pull request by babu-shanmugam**
Modified the osrmc_route in the library to accept a python dictionary object
which eases the usage of the route service from the python binding.

Also added a C++ class to parse the json return value to python dictionary so
that it is easy use in the python binding.

Tested against py2 as well py3

### CODE REVIEW

File: bindings/osrmcpy.py [hunk 1]
```diff
@@ -19,7 +19,8 @@
 
 class osrmc_error(c.c_void_p):
     def __str__(self):
-        return lib.osrmc_error_message(self)
+        return lib.osrmc_error_message(self).decode('latin1')
```
Comment (daniel-j-h) [inline bindings/osrmcpy.py:22]: Why is this needed?

File: bindings/osrmcpy.py [hunk 2]
```diff
@@ -59,7 +60,7 @@ def osrmc_error_errcheck(result, func, arguments):
 
 # Route
 lib.osrmc_route.restype = c.c_void_p
-lib.osrmc_route.argtypes = [c.c_void_p, c.c_void_p, c.c_void_p]
+lib.osrmc_route.argtypes = [c.c_void_p, c.py_object, c.c_void_p]
```
Comment (daniel-j-h) [inline bindings/osrmcpy.py:63]: I don't understand why we are passing a py_object to the C libosrmc library here. The C libosrmc library should not have a dependency on Python; it should be ABI and API stable and free-standing as much as possible.

For example when we use libosrmc from Haskell, how would we pass a py_object to the osrmc_route function?

**Review by daniel-j-h — CHANGES_REQUESTED** [review]

**Comment by daniel-j-h** [comment]
Making libosrmc depend on Python is not a way I want us to go.

libosrmc can and is used in other languages, for example from Rust, Haskell. There is no need to depend on Python when using libosrmc from these languages.

**Comment by babu-shanmugam** [comment]
I understand Daniel. Thanks for your time.

**Pull request closed without merging**